pkg/broadcaster/broadcaster.go

```diff
@@ -3,42 +3,37 @@ package broadcaster
 import (
 	"context"
 	"errors"
-	"fmt"
 	"io"
 	"strings"
 	"sync"
 	"time"
 )
 
 type client struct {
-	w    io.WriteCloser
+	w    io.Writer
 	msgs chan string
 	done chan struct{}
 }
 
 func (c *client) Close() error {
 	// TODO: Consider making the timeout configurable.
-	ctx, cancel := context.WithTimeout(context.Background(), 500*time.Millisecond) //nolint:mnd // Arbitrary duration.
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second) //nolint:mnd // Arbitrary duration.
```

Comment:
I think you can keep 500ms now, it doesn't matter as much as before. Plus 30s is a scary long time.